docsrc/hello_world.rst

```diff
@@ -44,6 +44,12 @@ given a set of N observations of i.i.d. binary data
 The :class:`CmdStanModel` class manages the Stan program and its corresponding compiled executable.
 It provides properties and functions to inspect the model code and filepaths.
 
+A `CmdStanModel` can be instantiated from a Stan file or its corresponding compiled executable file.
+When instantiated from a Stan file, the default behavior is to compile the Stan file,
+unless a newer executable file exists, in which case, an INFO message is logged.
```

Comment:
we could mention this - how much detail should go in the "Hello World" section?
we should definitely mention this somewhere.

Reply:
I think if we're going into the level of detail that it will not compile if a newer executable exists, we should also mention how to override this. IMO we could also just cut the sentence off after "compile the Stan file". Dealer's choice

Reply:
I  think we should kick the "INFO" message about not recompiling down to "DEBUG" status - new user was confused by this and thought it indicated a problem.
if we do that, then the "Hello World" description of model instantiation could be simpler.